### PR TITLE
Adding ER Diagrams to api schema

### PR DESCRIPTION
**Pull request by manasaV3**
Relates to

## Description
Adds ER diagram generation to the make command, such that the er diagram will be updated to reflect the changes to api schema. 

[Sample ER diagram](https://github.com/chanzuckerberg/cryoet-data-portal-backend/blob/363c64ca6aaefd3a01355b084a3b811d36d072c4/schema/api/v1.0.0/codegen/ER_DIAGRAM.md).

The dataset is set as root of the diagram, to help maintain consistency between different versions of the diagram.

### CODE REVIEW

File: schema/Makefile [hunk 1]
```diff
@@ -19,41 +20,61 @@ INGESTION_CONFIG_JSON_SCHEMA = ingestion_config/$(INGESTION_CONFIG_VERSION)/code
...
-.PHONY: clean-docs
-clean-docs:
-	rm -rf $(METADATA_DOCS)
+	rm -rf ingestion_config/latest
 
 .PHONY: build
-build: clean
+build: build-ingestion-config build-api
+
+# To build-ingestion-config for the non-default version, run `make build-api $INGESTION_CONFIG_VERSION=2.0.0 CORE_VERSION=2.0.0`
```
Comment (daniel-ji) [inline schema/Makefile:59]:
```
# To build-ingestion-config for the non-default version, run `make build-ingestion-config INGESTION_CONFIG_VERSION=v2.0.0 CORE_VERSION=v2.0.0`
```

File: schema/Makefile [hunk 2]
```diff
@@ -19,41 +20,61 @@ INGESTION_CONFIG_JSON_SCHEMA = ingestion_config/$(INGESTION_CONFIG_VERSION)/code
...
-	python3 schema.py materialize core/v2.0.0/metadata.yaml core/v2.0.0/common.yaml core/v2.0.0/codegen/metadata_materialized.yaml
-	python3 schema.py materialize api/v2.0.0/api_models.yaml core/v2.0.0/common.yaml api/v2.0.0/codegen/api_models_materialized.yaml
-
 	linkml generate pydantic --black $(METADATA_MATERIALIZED_YAML) > $(METADATA_MODELS_PY)
 	linkml generate pydantic $(INGESTION_CONFIG_MODELS_MATERIALIZED_YAML) > $(INGESTION_CONFIG_MODELS_PY)
 	linkml generate json-schema $(INGESTION_CONFIG_MODELS_MATERIALIZED_YAML) > $(INGESTION_CONFIG_JSON_SCHEMA)
 	cd ingestion_config && ln -s $(INGESTION_CONFIG_VERSION) latest && cd ..
 
+# To build-api for the non-default version, run `make build-api API_VERSION=2.0.0 CORE_VERSION=2.0.0`
+.PHONY: build-api
```
Comment (daniel-ji) [inline schema/Makefile:71]: Should it be `make build-api API_VERSION=v2.0.0 CORE_VERSION=v2.0.0`?

File: schema/Makefile [hunk 3]
```diff
@@ -19,41 +20,61 @@ INGESTION_CONFIG_JSON_SCHEMA = ingestion_config/$(INGESTION_CONFIG_VERSION)/code
...
+build-api: clean-core clean-api
+	# Hardcoding these while we wait for an upstream LinkML bug to be fixed: https://github.com/linkml/linkml-runtime/pull/330
```
Comment (daniel-ji) [inline schema/Makefile:73]: is this comment still needed?

**Review by daniel-ji — CHANGES_REQUESTED** [review]

File: schema/Makefile (hunk 3 above)
Comment (manasaV3) [reply schema/Makefile]: I believe not..

**Review by jgadling — APPROVED** [review]

**Review by daniel-ji — APPROVED** [review]
looks good to me!

**Pull request merged**